api/base/serializers.py

```diff
@@ -337,6 +338,36 @@ def get_attribute(self, obj):
         field_source_method = getattr(obj, self.source)
         return field_source_method(auth=auth)
 
+class AnonymizedRegexField(AuthorizedCharField):
+    """
+    Performs a regex replace on the content of the authorized object's
+    source field when an anonymous view is requested.
+
+    Example:
+        content = AnonymizedRegexField(source='get_content', regex='\[@[^\]]*\]\([^\) ]*\)', replace='@A User')
+    """
+
+    def __init__(self, source=None, regex=None, replace=None, **kwargs):
+        assert source is not None, 'The `source` argument is required.'
```

Comment:
I followed what the AuthorizedCharField had done, but yeah, it didn't seem necessary at all to have that.